controllers/operator/nmstate_controller.go

```diff
@@ -153,6 +155,11 @@ func (r *NMStateReconciler) applyRBAC(instance *nmstatev1.NMState) error {
 	data.Data["HandlerImage"] = os.Getenv("RELATED_IMAGE_HANDLER_IMAGE")
 	data.Data["HandlerPullPolicy"] = os.Getenv("HANDLER_IMAGE_PULL_POLICY")
 	data.Data["HandlerPrefix"] = os.Getenv("HANDLER_PREFIX")
+
+	if err := setClusterReaderExist(r.Client, data); err != nil {
```

Comment:
IMHO it would be cleaner to have this function only return if the cluster-reader-role exists (and an error eventually) and do the `data.Data["ClusterReaderExists"] = clusterReaderRoleExists` here, so we have it aligned with the other template settings. But that might be a personal opinion...

Reply:
Up to you, I started by that, but then it needs to check the error and do the assign, it looked less good imho
this way it is more compact, even that it is a bit less symmetric